Expose more serialized type metadata to IPC test API

achristensen07 opened the pull request:
#### cdc342c5f229f8c9c21c008c6bcb7d0edc46648a
<pre>
Expose more serialized type metadata to IPC test API
<a href="https://bugs.webkit.org/show_bug.cgi?id=271096">https://bugs.webkit.org/show_bug.cgi?id=271096</a>
<a href="https://rdar.apple.com/124721207">rdar://124721207</a>

Reviewed by Charlie Wolfe.

* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.messages.in:
* Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.messages.in:
* Source/WebKit/GPUProcess/webrtc/RemoteMediaRecorder.messages.in:
* Source/WebKit/Platform/IPC/ArgumentCoders.h:
(IPC::SimpleArgumentCoder::encode): Deleted.
(IPC::SimpleArgumentCoder::decode): Deleted.
* Source/WebKit/Platform/IPC/ObjectIdentifierReference.serialization.in:
* Source/WebKit/Scripts/webkit/messages.py:
(types_that_must_be_moved):
(serialized_identifiers):
(headers_for_type):
* Source/WebKit/Shared/Cocoa/CoreIPCCFType.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.serialization.in:
* Source/WebKit/Shared/Extensions/WebExtensionContextParameters.serialization.in:
* Source/WebKit/Shared/Extensions/WebExtensionTab.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in:
* Source/WebKit/Shared/TextCheckerState.h:
* Source/WebKit/Shared/TextFlags.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUExtent3D.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUOrigin2D.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUOrigin3D.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUOutOfMemoryError.serialization.in:
* Source/WebKit/Shared/cf/CoreIPCNumber.serialization.in:
* Source/WebKit/Shared/ios/DynamicViewportSizeUpdate.serialization.in:
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.messages.in:
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.messages.in:
* Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.messages.in:
* Source/WebKit/WebProcess/GPU/graphics/PrepareBackingStoreBuffersData.serialization.in:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSourceProviderManager.messages.in:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCAAnimationRemoteProperties.serialization.in:
* Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.messages.in:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IPCTestingAPI.mm:

Canonical link: <a href="https://commits.webkit.org/276225@main">https://commits.webkit.org/276225@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/61a4ddaf4ce28885ad592efd38afb6dad2c0dbfc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44078 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23147 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46511 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46720 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/40111 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46383 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27064 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20539 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36334 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44656 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20167 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/37957 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/17352 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/43955 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/17645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39057 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2128 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40237 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39346 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48304 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19057 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15620 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/43179 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/20427 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/41905 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20644 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6043 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/20054 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Source/WebKit/Shared/Cocoa/CoreIPCCFType.serialization.in:
```diff
@@ -29,3 +29,9 @@ webkit_platform_headers: "CoreIPCCFType.h"
+#if HAVE(SEC_KEYCHAIN)
+using WebKit::CFObjectValue = std::variant<std::nullptr_t, WebKit::CoreIPCCFArray, WebKit::CoreIPCBoolean, WebKit::CoreIPCCFCharacterSet, WebKit::CoreIPCData, WebKit::CoreIPCDate, WebKit::CoreIPCCFDictionary, WebKit::CoreIPCNull, WebKit::CoreIPCNumber, WebKit::CoreIPCString, WebKit::CoreIPCCFURL, WebKit::CoreIPCSecCertificate, WebKit::CoreIPCSecTrust, WebKit::CoreIPCCGColorSpace, WebCore::Color,  ...
+#else
+using WebKit::CFObjectValue = std::variant<std::nullptr_t, WebKit::CoreIPCCFArray, WebKit::CoreIPCBoolean, WebKit::CoreIPCCFCharacterSet, WebKit::CoreIPCData, WebKit::CoreIPCDate, WebKit::CoreIPCCFDictionary, WebKit::CoreIPCNull, WebKit::CoreIPCNumber, WebKit::CoreIPCString, WebKit::CoreIPCCFURL, WebKit::CoreIPCSecCertificate, WebKit::CoreIPCSecTrust, WebKit::CoreIPCCGColorSpace, WebCore::Color,  ...
```
Shouldn't `WebKit::CoreIPCSecAccessControl` be guarded by `HAVE(SEC_ACCESS_CONTROL)`?

achristensen07 replied:
HAVE_SEC_ACCESS_CONTROL is defined to 1 for all Cocoa platforms, so I simplified it here.  We don't have the ability yet to have #if inside the list of variant types, so if I included HAVE(SEC_ACCESS_CONTROL) I'd have to have 2 more using WebKit::CFObjectValue = ... that aren't compiled anywhere.

charliewolfe commented on Source/WebKit/WebProcess/GPU/graphics/PrepareBackingStoreBuffersData.serialization.in:
```diff
@@ -39,3 +39,11 @@ header: "PrepareBackingStoreBuffersData.h"
 #endif
+
+#if ENABLE(RE_DYNAMIC_CONTENT_SCALING) && ENABLE(GPU_PROCESS)
+using WebKit::ImageBufferBackendHandle = std::variant<WebCore::ShareableBitmap::Handle, MachSendRight, WebCore::DynamicContentScalingDisplayList>
```
Looks like `MachSendRight` needs `PLATFORM(COCOA)`?

achristensen07 replied:
Good catch

charliewolfe commented on Source/WebKit/WebProcess/GPU/graphics/PrepareBackingStoreBuffersData.serialization.in:
```diff
@@ -39,3 +39,11 @@ header: "PrepareBackingStoreBuffersData.h"
+#endif
+
+#if !ENABLE(RE_DYNAMIC_CONTENT_SCALING) && ENABLE(GPU_PROCESS)
+using WebKit::ImageBufferBackendHandle = std::variant<WebCore::ShareableBitmap::Handle, MachSendRight>
```
and here

achristensen07 committed:
Expose more serialized type metadata to IPC test API
https://bugs.webkit.org/show_bug.cgi?id=271096
rdar://124721207

Reviewed by Charlie Wolfe.

* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.messages.in:
* Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.messages.in:
* Source/WebKit/GPUProcess/webrtc/RemoteMediaRecorder.messages.in:
* Source/WebKit/Platform/IPC/ArgumentCoders.h:
(IPC::SimpleArgumentCoder::encode): Deleted.
(IPC::SimpleArgumentCoder::decode): Deleted.
* Source/WebKit/Platform/IPC/ObjectIdentifierReference.serialization.in:
* Source/WebKit/Scripts/webkit/messages.py:
(types_that_must_be_moved):
(serialized_identifiers):
(headers_for_type):
* Source/WebKit/Shared/Cocoa/CoreIPCCFType.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.serialization.in:
* Source/WebKit/Shared/Extensions/WebExtensionContextParameters.serialization.in:
* Source/WebKit/Shared/Extensions/WebExtensionTab.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in:
* Source/WebKit/Shared/TextCheckerState.h:
* Source/WebKit/Shared/TextFlags.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUExtent3D.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUOrigin2D.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUOrigin3D.serialization.in:
* Source/WebKit/Shared/WebGPU/WebGPUOutOfMemoryError.serialization.in:
* Source/WebKit/Shared/cf/CoreIPCNumber.serialization.in:
* Source/WebKit/Shared/ios/DynamicViewportSizeUpdate.serialization.in:
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.messages.in:
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.messages.in:
* Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.messages.in:
* Source/WebKit/WebProcess/GPU/graphics/PrepareBackingStoreBuffersData.serialization.in:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSourceProviderManager.messages.in:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCAAnimationRemoteProperties.serialization.in:
* Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.messages.in:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IPCTestingAPI.mm:

Canonical link: https://commits.webkit.org/276225@main
